Lib/os.py

```diff
@@ -974,14 +974,15 @@ def spawnlpe(mode, file, *args):
 # command in a shell can't be supported.
 if sys.platform != 'vxworks':
     # Supply os.popen()
-    def popen(cmd, mode="r", buffering=-1):
+    def popen(cmd, mode="r", buffering=-1, encoding=None):
         if not isinstance(cmd, str):
             raise TypeError("invalid cmd type (%s, expected string)" % type(cmd))
         if mode not in ("r", "w"):
             raise ValueError("invalid mode %r" % mode)
         if buffering == 0 or buffering is None:
             raise ValueError("popen() does not support unbuffered streams")
         import subprocess, io
+        encoding = io.text_encoding(encoding)
```

Comment:
Maybe I'm missing something obvious, but shouldn't this be passed to subprocess.Popen?

Reply:
Thank you. I was too rushed.
I will fix it in follow up PR.
https://github.com/python/cpython/pull/92415